Tella/Scenes/Authentication/Views/Unlock/UnlockView.swift

```diff
@@ -0,0 +1,162 @@
+    @Environment(\.presentationMode) var presentationMode: Binding<PresentationMode>
+    @State private var cancellable: Set<AnyCancellable> = []
+    @State private var isLoading : Bool = false
+    var type : PasswordTypeEnum
+    var body: some View {
+        ContainerView {
+            VStack(alignment: .center) {
+                Spacer(minLength: 30)
+                
+                Image("tella.logo")
+                    .frame(width: 65, height: 72)
+                    .aspectRatio(contentMode: .fit)
+                
+                Spacer(minLength: 23)
+                
+                Text(titleString)
+                    .font(.custom(Styles.Fonts.semiBoldFontName, size: 18))
+                    .foregroundColor(.white)
+                    .lineSpacing(7)
+                    .multilineTextAlignment(.center)
+                    .padding(EdgeInsets(top: 0, leading: 67, bottom: 0, trailing: 67))
+                
+                Spacer()
+                
+                if(viewModel.shouldShowAttemptsWarning) {
+                    Text(viewModel.warningText())
+                        .font(.custom(Styles.Fonts.regularFontName, size: 14))
+                        .foregroundColor(.white)
+                        .lineSpacing(7)
+                        .multilineTextAlignment(.center)
+                        .padding(EdgeInsets(top: 0, leading: 67, bottom: 0, trailing: 67))
+                    Spacer()
+                }
+                
+                
+
+                if(type == .tellaPassword) {
+                    PasswordTextFieldView(fieldContent: $viewModel.loginPassword,
+                                          isValid: .constant(true),
+                                          shouldShowError: $viewModel.shouldShowUnlockError) {
+                        loginActions()
+                    }
+                } else {
+                    PasswordTextFieldView(fieldContent: $viewModel.loginPassword,
+                                          isValid: .constant(true),
+                                          shouldShowError: $viewModel.shouldShowUnlockError,
+                                          disabled: true)
+                    
+                    Spacer(minLength: 20)
+                    
+                    PinView(fieldContent: $viewModel.loginPassword,
+                            keyboardNumbers: UnlockKeyboardNumbers) {
+                        loginActions()
+                    }
+                }
+                
+                
+                
+                Spacer()
+            }
+            
+            if  isLoading {
+                CircularActivityIndicatory()
+            }
+            
+        }
+        
+        .overlay(lockChoiceView)
+        .onAppear {
+            viewModel.initUnlockData()
+        }
+    }
+    
+    var titleString : String {
+        if(type == .tellaPin) {
+            if viewModel.shouldShowUnlockError {
```

Comment:
@ValbuenaG Could you put this code in a var that returns the pin or password title string and use switch case